django/db/migrations/operations/utils.py

```diff
@@ -14,6 +14,31 @@ def is_referenced_by_foreign_key(state, model_name_lower, field, field_name):
     return False
 
 
+def describe_operation(operation, backwards):
+    """
+    Returns a string, intended for `stdout.write`, that describes
+    a migration operation.
+    """
+    if hasattr(operation, "code"):
+        action = operation.code.__doc__ if backwards is not True else operation.reverse_code.__doc__
+        prefix = ""
+    elif hasattr(operation, "sql"):
+        action = operation.sql if backwards is not True else operation.reverse_sql
+        prefix = ""
+    else:
+        action = ""
+        prefix = "Undo " if backwards is True else ""
+    if action is None:
+        error = True
+        action = "IRREVERSIBLE"
+    else:
+        action = "{}".format(action.replace("\n", ""))
+        error = False
+    action = action if len(action) < 40 else '{}...'.format(action[:37])
+    styled_action = " -> {}".format(action) if action else action
+    return '{}{}{}'.format(prefix, operation.describe(), styled_action), error
```

Comment:
How about: `prefix + operation.describe() + styled_action` instead of `format()`? I saw some other places that could use that style as well.

Reply:
With this and other comments, it seems the preferred pattern is to concatenate strings with `+` rather than using the `format()` thing. Is there something in the contribution guidelines I missed perhaps ?  I do see `.format()` used ubiquitously throughout the codebase.

I'm happy to swap all the formatting stuff over to `string1 + string2`. Will fix that for sure, and would be interested in the reasoning. Other than stylistic preference, is there a difference here I should be aware of ?

Reply:
String concatenation looks simpler to me for these cases. `format()` is somewhat verbose for me.